pkg/k8shandler/cr.go

```diff
@@ -0,0 +1,24 @@
+package k8shandler
+
+import (
+	"context"
+	"fmt"
+
+	loggingv1 "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func FindElasticserachCR(c client.Client, ns string) (*loggingv1.Elasticsearch, error) {
+	esl := &loggingv1.ElasticsearchList{}
+	opts := &client.ListOptions{Namespace: ns}
+
+	if err := c.List(context.TODO(), opts, esl); err != nil {
+		return nil, fmt.Errorf("failed to find elasticsearch instance in %q: %s", ns, err)
+	}
+
+	if len(esl.Items) == 0 {
+		return nil, fmt.Errorf("failed to find elasticsearch instance in %q: empty result set", ns)
+	}
+
+	return &esl.Items[0], nil
```

Comment:
correct, technically we can have any number of ES crs in a namespace... the EO will resolve each of them to their own cluster. whether or not that is a good idea is another discussion...

Reply:
As talked on the line with @jcantrill , since our kibana CR is bound to a single ES in the same namespace (See how we build the kibana pod spec's ELASTICSEARCH_HOSTS), we keep it as it is for now. This is a good amendment to amend the kibana CR to have an endpoint URL so that CLO can create both CRs (ES & Kibana) and configure them without much coupling.

Reply:
separate from this PR: maybe as tech debt we should consider having EO only allow a single ES per namespace